File: 0009-palindrome-number/0009-palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if(x<0)
         return false;
        int rev =0,n=x;
        while(x!=0)
        {   
            if(rev >INT_MAX/10 || rev<INT_MIN/10)
               return false;
            rev = (rev*10)+(x%10);
            x/=10;
        }
        return rev==n;
    }
};

File: 0009-palindrome-number/0009-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        int oriX = x, rev =0;
        if(x<0)
         return false;

        while(x!=0)
        {
            rev = rev*10+(x%10);
            x/=10;
        }
        return oriX==rev;
    }
}